modules/@apostrophecms/rich-text-widget/ui/apos/tiptap-extensions/Heading.js

```diff
@@ -0,0 +1,39 @@
+// Lock Heading levels down to just those provided via configuration
+import Heading from '@tiptap/extension-heading';
+
+export default (options) => {
+  const headings = options.styles.filter(style => style.type === 'heading');
+  const levels = headings.map(heading => heading.options.level);
+  const defaultLevel = headings.filter(heading => heading.def).length
+    ? headings.filter(heading => heading.def)[0].options.level
+    : levels[0];
+  // Map of headings to their attributes
+  const attrs = {};
+  levels.forEach(level => {
+    attrs[level] = headings.find(heading => heading.options.level === level).options.attrs || {};
+  });
+  return Heading.extend({
+    defaultOptions: {
+      levels,
+      attrs
+    },
+    addAttributes() {
+      return {
+        level: {
+          default: defaultLevel,
```

Comment:
puts a finer point on _which_ heading should be default, the part 2 of the Document subclass

Reply:
^ This point is less relevant now that we make a special node for our default, but it still makes sense to hem headings in based on their configuration.